### PR TITLE
add `scheme` and `scheme_warning` to LinkDestination

### PR DESCRIPTION
**Pull request by Simon-Laux**

### CODE REVIEW

**Review by WofWca — APPROVED** [review]

**Comment by dotlambda** [comment]
I don't think having such an extensive whitelist is a good idea. If it were just a very short list (e.g. http, https, mailto) people would be less likely to ask for additions to the list.

File: src/parser/link_url.rs [hunk 1]
```diff
@@ -41,9 +48,48 @@ pub struct PunycodeWarning {
...
+        // Decentralized web
+        | "ipfs" | "ipns" | "dweb" | "dat" | "ssb"
+        // crypto curencies
+        | "bitcoin" | "bitcoincash" | "eth" | "ethereum"
+        // payment
+        | "payto" /* https://datatracker.ietf.org/doc/html/rfc8905, but has probably not high adoption */ | "taler"
+        // debian apps
+        | "apt"
+        // torrent
+        | "magnet"
```
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply src/parser/link_url.rs:91]: that list is made up, I skimmed over the Dana list of all registered uri schemes and some other sources and picked the ones I deemed safe-ish. We can remove the file Uris again (ipfs and dat) if we think there is any chance people use a risky implementation that opens up executables like `.jar` or `.desktop` files without additional user confirmation.

[the remark this answers is not in the document]
Comment (Simon-Laux) [reply src/parser/link_url.rs:91]: Torrent clients usually have more dialogs and don't usually start an executable file they downloaded, or am I wrong on that assumption?

**Comment by Simon-Laux** [comment]
> I don't think having such an extensive whitelist is a good idea. If it were just a very short list (e.g. http, https, mailto, tel) people would be less likely to ask for additions to the list.

I don't think that's a big problem, they just need to guarantee the uri scheme they are proposing is handled safely by its applications and does not allow for remote code execution without user consent.

File: src/parser/link_url.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (WofWca) [reply src/parser/link_url.rs:91]: Usually yes, but the point is, IMU, you could say the same about Samba clients, and you'd be wrong.

[the remark this answers is not in the document]
Comment (Simon-Laux) [reply src/parser/link_url.rs:91]: for samba it is known, I don't know a torrent client that does this. So I agree in principle but in praxis I have not seen this issue with torrent clients yet. Though I don't have used torrents very often, so could be that I just not know many torrent clients.

**Comment by Simon-Laux** [comment]
we changed desktop to only open http/s links, open mailto links in itself and offer to copy any other scheme. So this pr is not needed anymore.
#44 is only the part of this pr of linkify-ing new uri schemes.

**Pull request closed without merging**